snap/validate.go

```diff
@@ -47,6 +48,85 @@ func ValidateEpoch(epoch string) error {
 	return nil
 }
 
+// ValidateLicense checks if a string is a valid SPDX expression.
+func ValidateLicense(license string) error {
```

Comment:
String peeks is how one builds an AST, so that's not an issue on itself.

That said, this function is clearly on the loose end. Someone needs to implement a proper parser basing off of the SPDX specification itself.